### PR TITLE
Updated deep loss CNV calls 

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Add logic for a "deep loss" call for focal CN output files.

#### What was your approach?
Changed:
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/6bf301eea42a7f8812c3b14cee35a961097f8ecf/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L240-L247

To:
```

# Merge and annotated no X&Y
autosome_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_no_xy,
                                                   txdb_exons = tx_exons) %>%
  # mark possible amplifications and deep loss in autosomes
  dplyr::mutate(status = dplyr::case_when(
    copy_number > (2 * ploidy) ~ "amplification",
    copy_number == 0 ~ "deep loss",
    TRUE ~ status
  ))

```
Changed:
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/6bf301eea42a7f8812c3b14cee35a961097f8ecf/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L265-L267

To:
```
  sex_chrom_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_sex_chrom,
                                                      txdb_exons = tx_exons) %>%
  # mark possible deep loss in sex chromosome
  dplyr::mutate(status = case_when(
    copy_number == 0  ~ "deep loss",
   TRUE ~ status
  ))
```

#### What GitHub issue does your pull request address?
#1099

#### Which areas should receive a particularly close look?
 - Please verify in the conditions for "deep loss" seem right to you.
 - The chromosome arm and cytoband level copy_number annotation uses only "gain" and "loss" as status; [amplification is converted to gain](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/aee3127e537ae8fde80aba5a754774c9c3803343/analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd#L92-L94) 
 Should we convert the "deep loss" added in this PR as "loss" for cytoband annotation OR do we want to add amplification and deep loss annotation to cytobands?

#### Is there anything that you want to discuss further?
NA

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
table

#### What is your summary of the results?
"Deep loss" calls were added to focal CN files for autosomes and X+Y calls.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by kgaonkar6** [comment]
@jharenza 
I’ve now updated the CN==0 as the only criteria for “deep loss” 

>The chromosome arm and cytoband level copy_number annotation uses only "gain" and "loss" as status; amplification is converted to gain
Should we convert the "deep loss" added in this PR as "loss" for cytoband annotation OR do we want to add amplification and deep loss annotation to cytobands?
Does this sound ok as is  ? 

XY chromosomes don’t have a logic in the current code y to add applications does that sound ok too?

File: analyses/focal-cn-file-preparation/04-prepare-cn-file.R [hunk 1]
```diff
@@ -240,9 +240,10 @@ cnv_no_xy <- cnv_df %>%
 # Merge and annotated no X&Y
 autosome_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_no_xy,
                                                    txdb_exons = tx_exons) %>%
-  # mark possible amplifications in autosomes
+  # mark possible amplifications and deep loss in autosomes
   dplyr::mutate(status = dplyr::case_when(
     copy_number > (2 * ploidy) ~ "amplification",
+    copy_number == 0 ~ "deep loss",
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/04-prepare-cn-file.R:246]:
```suggestion
    copy_number == 0 ~ "deep deletion",
```

just so we don't have to change it later, updating to this standard nomenclature

File: analyses/focal-cn-file-preparation/04-prepare-cn-file.R [hunk 2]
```diff
@@ -264,7 +265,13 @@ if (xy_flag) {
 
   # Merge and annotated no X&Y
   sex_chrom_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_sex_chrom,
-                                                      txdb_exons = tx_exons)
+                                                      txdb_exons = tx_exons) %>%
+  # mark possible deep loss in sex chromosome
+  dplyr::mutate(status = dplyr::case_when(
+    copy_number == 0  ~ "deep loss",
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/04-prepare-cn-file.R:271]:
```suggestion
    copy_number == 0  ~ "deep deletion",
```

File: analyses/focal-cn-file-preparation/README.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ See the notebook for more information. This notebook also prepares lists of copy
   | biospecimen_id | status | copy_number | ploidy | ensembl | gene_symbol | cytoband |
   |----------------|--------|-------------|--------|---------|-------------|---------|
   Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number > (2 * ploidy) are marked as amplifications](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e2058dd43d9b1dd41b609e0c3429c72f79ff3be6/analyses/focal-cn-file-preparation/03-prepare-cn-file.R#L275) in the `status` column.
+  Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number <= (ploidy -2 ) in autosomes chromosomes](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L244-L247) and [copy_number==0 in X/Y chromosomes are marked as deep losses](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L270-L272) in the `status` column.
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/README.md:49]:
```suggestion
  Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number <= (ploidy -2 ) in autosomes chromosomes](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L244-L247) and [copy_number==0 in X/Y chromosomes are marked as deep deletions](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L270-L272) in the `status` column.
```

**Review by jharenza — APPROVED** [review]
This looks good, just want to switch the nomenclature to `deep deletion`

**Comment by kgaonkar6** [comment]
Thanks for the review @jharenza 🙌

**Comment by jharenza** [comment]
> The chromosome arm and cytoband level copy_number annotation uses only "gain" and "loss" as status; amplification is converted to gain

For this, we should remove the following lines to retain amplifications:

https://github.com/kgaonkar6/OpenPBTA-analysis/blob/aee3127e537ae8fde80aba5a754774c9c3803343/analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd#L92-L94

**Review by jharenza — CHANGES_REQUESTED** [review]
see comment about removing the amp-->gain change in XY seg file

**Review by jharenza — APPROVED** [review]

File: analyses/focal-cn-file-preparation/README.md [hunk 4]
```diff
@@ -46,6 +46,7 @@ See the notebook for more information. This notebook also prepares lists of copy
   | biospecimen_id | status | copy_number | ploidy | ensembl | gene_symbol | cytoband |
   |----------------|--------|-------------|--------|---------|-------------|---------|
   Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number > (2 * ploidy) are marked as amplifications](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e2058dd43d9b1dd41b609e0c3429c72f79ff3be6/analyses/focal-cn-file-preparation/03-prepare-cn-file.R#L275) in the `status` column.
+  Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number <= (ploidy -2 ) in autosomes chromosomes](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L244-L247) and [copy_number==0 in X/Y chromosomes are marked as deep deletions](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/5e9b68ed0c37d6c87bc67ee1f95be9e47ec9b4ca/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L270-L272) in the `status` column.
```
Comment (jaclyn-taroni) [inline analyses/focal-cn-file-preparation/README.md:49]: Based on my read of `analyses/focal-cn-file-preparation/04-prepare-cn-file.R`, this should be:

```suggestion
  Any segment that is copy neutral is filtered out of this table. In addition, [any segments with copy number > (2 * ploidy) are marked as amplifications](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e2058dd43d9b1dd41b609e0c3429c72f79ff3be6/analyses/focal-cn-file-preparation/03-prepare-cn-file.R#L275) in the `status` column. 
Any segments with copy number == 0  are marked as deep deletions in the `status` column.
```

**Review by jharenza — APPROVED** [review]

**Pull request merged**